rust/lance/src/index.rs

```diff
@@ -707,6 +707,28 @@ impl DatasetIndexExt for Dataset {
             });
         }
 
+        if index_name == FRAG_REUSE_INDEX_NAME {
+            let index = self
+                .open_frag_reuse_index(&NoOpMetricsCollector)
+                .await?
+                .unwrap();
```

Comment:
Should we handle the "no frag reuse index" case better?  Maybe return a specific error message like "there is no fragment reuse index"?